src/plugins/footnotes/index.ts

```diff
@@ -90,6 +102,108 @@ export const uncitedFootnoteWidget = () => () => {
   element.innerHTML = alertIcon
   return element
 }
+interface inlineFootnote {
+  node: InlineFootnoteNode | null
+  pos: number | null
+}
+
+const getInlineFootnote = (view: EditorView, id: string): inlineFootnote => {
+  let inlineFootnote: InlineFootnoteNode | null = null
+  let footnotePosition: number | null = null
+
+  view.state.doc.descendants((node, pos) => {
+    const footnote = node as InlineFootnoteNode
+
+    if (footnote.attrs.rids?.includes(id)) {
+      inlineFootnote = footnote
+
+      footnotePosition = pos
+      return false
+    }
+  })
+
+  return { node: inlineFootnote, pos: footnotePosition }
```

Comment:
I would expect a returned array that includes all the related inline-footnotes not just the first occurrence.